buf.gen.yaml

```
@@ -0,0 +1,13 @@
+# The version of the generation template.
+# Required.
+# The only currently-valid value is v1beta1.
+version: v1beta1
+
+# The plugins to run.
+plugins:
+  # The name of the plugin.
+  - name: gogofaster
+    # The the relative output directory.
+    out: proto
+    # Any options to provide to the plugin.
+    opt: Mgoogle/protobuf/timestamp.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/duration.proto=github.com/golang/protobuf/ptypes/duration,plugins=grpc,paths=source_relative
```

Some suggestions to make the comments more explanatory:

```suggestion
plugins:
  - name: gogofaster
    # The output directory relative to the source files.
    out: proto
    # Set options to assign import paths to the well-known types
    # and to enable service generation.
    opt: Mgoogle/protobuf/timestamp.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/duration.proto=github.com/golang/protobuf/ptypes/duration,plugins=grpc,paths=source_relative
```

I'm actually not sure if the `out` option is relative to the source files (as my suggestion says) or to the repository root. Whichever it is, I think we should say. (The proto plugin is set to be `source_relative`, but I don't know if this is an option for the compiler or the plugin)

---

`out` is relative to where the generate command is run. noted